src/spikeinterface/core/datasets.py

```diff
@@ -59,7 +66,7 @@ def download_dataset(repo=None, remote_path=None, local_folder=None, update_if_e
 
     if remote_path is None:
         warnings.warn(message="No remote path provided, returning None")
-        return
+        return None  # TODO: is this the best option? I Think this should not accept remote_path is None.
```

Comment:
Not sure, but let's just add typing in this PR (I'll leave the comment in)

Reply:
Yes, I find accepting remote_path as None very strange as the function just kicks you out and throws a warning.

Probably was done as a hack for something else at some point though.

Reply:
Should we just raise an exception?

Reply:
Yes, I think would be better.

But I think it would be even better if `remote_path` would be a required argument in the signature. 

The `repo` keyword in the signature is set automatically to:

https://github.com/SpikeInterface/spikeinterface/blob/0468095ed194755ebec26fa22e8ec814c271ce64/src/spikeinterface/core/datasets.py#L46-L49

So, it makes sense to have `repo` as an optional argument. 

But the `remote_path` keyword is determines the dataset that you want to download. I don't see why it makes sense to pass no dataset. 

I think we should the warning to a deprecation warning and then at some point make `remote_path` a required argument.